src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,1001 @@ static ColliderCylinderInit D_809D5C40 = {
+        sIsEffectsInitialized = 1;
+        globalCtx->specialEffects = &sClearTagEffects[0];
+        for (i = 0; i < CLEAR_TAG_EFFECT_MAX_COUNT; i++) {
+            sClearTagEffects[i].type = CLEAR_TAG_EFFECT_AVAILABLE;
+        }
+        this->drawMode = CLEAR_TAG_DRAWMODE_ALL;
+    }
+}
+
+/**
+ * Calculate a floor tangent.
+ * This is used for the ground flash display lists and Arwing shadow display lists to snap onto the floor.
+ */
+void EnClearTag_CalculateFloorTangent(EnClearTag* this) {
+    f32 x;
+    f32 y;
+    f32 z;
+
+    // If there is a floor poly below the Arwing, calculate the floor tangent.
+    if (this->actor.floorPoly != NULL) {
+        x = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.x);
+        y = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.y);
+        z = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.z);
+
+        this->floorTangent.x = -Math_FAtan2F(-z * y, 1.0f);
+        this->floorTangent.z = Math_FAtan2F(-x * y, 1.0f);
+    }
+}
+
+/**
+ * EnClear_Tag update function.
+ *
+ * ClearTag has three variants:
+ * CLEAR_TAG_CUTSCENE_ARWING which is the regular Arwing and plays a cutscene on spawn.
+ * CLEAR_TAG_ARWING which is a regular Arwing.
+ * CLEAR_TAG_LASER which are the lasers fired by the Arwing.
+ *
+ * This function controls the Arwing flying. A target position is set and the Arwing flies toward it based on
+ * calculated directions from the current position.
+ * This function fires Arwing lasers when the state is CLEAR_TAG_STATE_TARGET_LOCKED.
+ * This function controls the cutscene that plays when the Arwing has params for
+ * cutscene. The cutscene stops playing when the Arwing is a specified distance from the starting point.
+ *
+ * work[0] for Arwing in locked on mode is a timeout when the Arwing reverts back to flying mode.
+ * work[0] for Arwing in fly mode is a timeout for when you can enter locked on mode.
+ * work[0] for lasers is a cutscene timer.
+ * work[1] for Arwing is a timeout for when it will enter locked on mode.
+ * work[2] for Arwing is a cutscene timer.
+ */
+void EnClearTag_Update(Actor* thisx, GlobalContext* globalCtx) {
+    u8 hasAtHit = 0;
+    s16 i;
+    s16 xRotationTarget;
+    s16 rotationScale;
+    EnClearTag* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
+    Player* player = PLAYER;
+    f32 vectorToTargetX;
+    f32 vectorToTargetY;
+    f32 vectorToTargetZ;
+    s16 zWorldRotationTarget;
+    s16 cutsceneTimer;
+    f32 loseTargetLockDistance;
+    s32 pad;
+    s16 yWorldRotationTarget;
+    s16 xWorldRotationTarget;
+    f32 targetCircleX;
+    f32 targetCircleZ;
+    f32 cutsceneCameraCircleX;
+    f32 cutsceneCameraCircleZ;
+    s16 isArwingAlive;
+    Vec3f cutsceneCameraAtTarget;
+    Vec3f cutsceneCameraEyeTarget;
+    Vec3f crashEffectLocation;
+    Vec3f crashEffectVelocity;
+    Vec3f debrisEffectAcceleration;
+
+    this->timer++;
+
+    if (this->drawMode != CLEAR_TAG_DRAWMODE_EFFECT) {
+        // Decrement the "work" timers.
```

Comment:
imo comment not needed, you can quickly see thats whats happening

Reply:
Hmmm. Yeah I think that's fair.